unlock-app/src/components/creator/members/ExtendKeysDrawer.tsx

```diff
@@ -0,0 +1,240 @@
+      },
+      () => void 0
+    )
+  }
+
+  const extendKeyMutation = useMutation(onExtendKeys)
+
+  const onExtendDuration = async () => {
+    const isFormValid = await trigger()
+    const { expiration, neverExpires } = getValues()
+    if (isFormValid) {
+      const prevExpiration = dayjs(parseInt(currentExpiration) * 1000)
+      const newExpiration = dayjs(expiration)
+      const timeDiffFromNow = newExpiration.diff(prevExpiration, 'second')
+
+      const extendDuration = neverExpires ? MAX_UINT : timeDiffFromNow
+
+      if (extendDuration > 0 || extendDuration === MAX_UINT) {
+        const keyMutationPromise = extendKeyMutation.mutateAsync({
+          extendDuration,
+          lockAddress,
+          tokenId,
+        })
+        await ToastHelper.promise(keyMutationPromise, {
+          loading: 'Extending key duration',
+          success: 'Keys duration extended',
+          error: 'Error with extending keys',
+        })
+        onComplete()
+        reset(defaultValues)
+      } else {
+        ToastHelper.error(`Expiration date can't be in the past`)
+      }
+    }
+  }
+  const { neverExpires } = getValues()
+  const loading = extendKeyMutation.isLoading
+  return (
+    <form
+      className="flex flex-col w-full gap-3"
+      onSubmit={handleSubmit(onExtendDuration)}
+    >
+      <Input
+        type="datetime-local"
+        placeholder="Key expiration date"
+        label="Key expiration date"
+        disabled={neverExpires || loading}
+        min={1}
+        {...register('expiration')}
+      />
+
+      <div>
+        <label htmlFor="never-expires">
+          Never Expires
+          <input
+            id="never-expires"
+            className="ml-2 align-middle"
+            type="checkbox"
+            disabled={loading}
+            {...register('neverExpires', {
+              onChange: () => {
+                setValue('expiration', '')
+              },
+            })}
+          />
+        </label>
+      </div>
+      <Button disabled={!isDirty || loading} type="submit">
+        <div className="flex items-center gap-2">
+          {extendKeyMutation.isLoading && (
+            <Spinner className="mr-1 animate-spin" />
+          )}
+          <span>
+            {!extendKeyMutation.isLoading
+              ? 'Extend key duration'
+              : 'Extend key duration...'}
+          </span>
+        </div>
+      </Button>
+    </form>
+  )
+}
+
+export const ExtendKeysDrawer = ({
+  isOpen,
+  setIsOpen,
+  selectedKey,
+  refetch,
+}: ExtendKeyDrawerProps) => {
+  const owner = selectedKey?.owner
+  const addressToEns = useEns(owner!)
+
+  const {
+    lockAddress,
+    tokenId,
+    expiration: currentExpiration,
+  } = selectedKey ?? {}
+
+  if (!lockAddress && !tokenId) return null
+
+  const onComplete = async () => {
+    setIsOpen(false)
+    if (typeof refetch === 'function') {
+      await refetch()
+      // force reload to show keys in defaut list
```

Comment:
No... I have told you in the past, a full page reload is *never* the right solution,.

Reply:
Why do you need to reload? If it is just to update the value of the expiration, just update it from what the user entered?

Reply:
When we extend expired key, generally we are filtering by `Expired -> expired`, need to go back to the previous list to see the key, or change filters because it will no be present with the actual filter

Reply:
Ill change it with a nicer way

Reply:
@julien51 updated!